src/components/Loaders/Loaders.tsx

```diff
@@ -1,43 +1,83 @@
-import { useEffect, useState, Children, useRef } from "react";
-import LoadersData from '../../constants/getLoadersData';
-import Loader from "../../Interfaces/Loader";
-import Modal from "../Modal";
+import { useEffect, useState, Children, useRef } from 'react';
+import LoadersData, { LoaderCircle } from '../../constants/getLoadersData';
+import Loader from '../../Interfaces/Loader';
+import { LdsAnimationDirections } from '../../LdsAnimationDirections';
+import { LoaderService } from '../../LoaderService';
+import Modal from '../Modal';
 import './style.css';
 
 const LoadersComponent = () => {
   const [loaders, setLoaders] = useState<Loader[] | []>([]);
   const [modal, setModal] = useState<boolean>(false);
   const [selectedLoader, setSelectedLoader] = useState<Loader>();
   const loaderContainerRef = useRef<HTMLDivElement>(null);
+  const loaderService = new LoaderService();
 
-  useEffect(() => {
-    const styleEle = document.createElement("Style");
-    styleEle.innerText = LoadersData.map(loaderData => loaderData.cssRules).join("");
+  function goToService(selectedLoader: Loader) {
+    const newLoaderInfo: Loader = loaderService.configureLdsCircle(
+      selectedLoader,
+      'red',
+      '80px',
+      LdsAnimationDirections.lrt
+    );
+    const transformLoaders = loaders.map((loader: Loader) => {
+      if (newLoaderInfo.id === loader.id) {
+        return newLoaderInfo;
+      }
+      return loader;
+    });
+    switch (selectedLoader.id) {
+      case LoaderCircle.id:
+        console.log(transformLoaders);
+        break;
+      default:
+        console.log('default works');
+    }
+  }
+  function applyStyles() {
+    const styleEle = document.createElement('Style');
+    styleEle.innerText = loaders
+      .map((loaderData) => loaderData.cssRules)
+      .join('');
     if (loaderContainerRef && loaderContainerRef.current) {
       loaderContainerRef.current.append(styleEle);
     }
-    setLoaders([...LoadersData])
-  }, [])
-  const showLoaderDetails = (data:Loader) => {
-    setSelectedLoader({...data})
-    setModal(!modal)
   }
+  useEffect(() => {
+    setLoaders([...LoadersData]);
+  }, []);
+  useEffect(() => {
+    applyStyles();
+  }, [loaders]);
```

Comment:
Remove `loaders` from this dependency array since we're not using it inside the `useEffect` block